api/handler/v1beta1/adapter.go

```diff
@@ -442,14 +443,24 @@ func ToResourceProto(spec models.ResourceSpec) (*pb.ResourceSpecification, error
 }
 
 func FromResourceProto(spec *pb.ResourceSpecification, storeName string, datastoreRepo models.DatastoreRepo) (models.ResourceSpec, error) {
+	if spec == nil {
```

Comment:
@irainia cmiiw, because these flows are not handled, it is panicing and server is crashing is it? Just want to check ideally the panic should just kill the goroutine handling the request not the entire server, can we cross check on this, as this means this issue can happen even in the future for all unhandled scenarios.

Reply:
@sravankorumilli , yes, it cashed because of panic. But, we didn't explicitly call any go routine to run it (such as [this](https://github.com/odpf/optimus/blob/e72edc7ef46cda9dd30691a5444f98f68e8bb1a7/api/handler/v1beta1/resource.go#L38)).

Reply:
@irainia we can use middleware as per [this](https://github.com/grpc-ecosystem/go-grpc-middleware/blob/master/recovery/examples_test.go) to recover from panics

Reply:
updated and tested based on the suggestion